### ISSUE TITLE
[Azure] Azure SDK for Go, update ACI

### ISSUE THREAD

**Issue by radu-matei** [virtual-kubelet/virtual-kubelet#44]
Hi, everyone!
First of all, this is a really awesome project, congrats!

Now I have a couple of questions regarding the Azure Container Instance provider:

- why did you write your own client for ACI and not use the SDK?
- as far as I last tried, ACI supports update - I am currently working on a sample and would love to contribute here 😄 

Thanks,
Radu M

**Comment by robbiezhang** [comment]
ACI supports "update" recently, after the current version virtual-kubelet is implemented. It's on our backlog to support "update" in virtual-kubelet, but you are welcome to contribute.

We didn't use the existing SDK for ACI because the available SDKs are based on api-version 2017-10-01-preview. It doesn't support secret volume, which is very important for the virtual-kubelet. So, we used the autorest to generate the go SDK upon api-version 2017-12-01-preview by ourselves. 
BTW, we notice that the generated code is not go-friendly. For example, the string array is translated to *(string[]), instead of (*string)[], which requires a lot of conversion when we use it in code. So, we made a few changes, and will report back to the auto-rest project.
